urlaa914a1de2_xu_hao_erilex_tgz-pJ8-erilex_esl_ESLJ8/src/erilex/tree/TreeBuildingHandler.java

```diff
@@ -0,0 +1,125 @@
+/*
+Copyright 2009 Hao Xu
+ertranne@hotmail.com
+
+This file is part of EriLex.
+
+EriLex is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 2 of the License, or
+(at your option) any later version.
+
+EriLex is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with EriLex; if not, write to the Free Software
+Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+package erilex.tree;
+
+import erilex.data.generic.Tree;
+import erilex.Handler;
+import erilex.NonTerminal;
+import erilex.Production;
+import erilex.Utils;
+import erilex.data.CharStream;
+import erilex.data.CharStream.Label;
+import java.util.Stack;
+
+/**
+ *
+ * @author ertri
+ */
+public class TreeBuildingHandler implements Handler {
+
+    Stack<Stack<Tree<ASTValueData>>> stack = new Stack();
+    Stack<Tree<ASTValueData>> top = new Stack<Tree<ASTValueData>>();
+    Tree<ASTValueData> ast;
+    public boolean longText = false;
+
+    public void handle(Production production, CharStream cs, Label start, Label finish) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 16. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 46-94 into a separate method.